lib/language_pack/ruby.rb

```diff
@@ -108,9 +108,13 @@ def compile
       end
       config_detect
       best_practice_warnings
+      warn_outdated_ruby
       cleanup
       super
     end
+  rescue => e
+    warn_outdated_ruby
```

Comment:
`warn_outdated_ruby` twice?

Reply:
I could maybe add a comment here. But in the event that a build fails the result might be because you're using old Ruby versions and upgrading them might help to solve your issue. The `warn_outdated_ruby` method is memoized so that it will only fire once.

Essentially if your project makes it to the first `warn_outdated_ruby` invocation and then later gets an error, it won't fire again. If it gets an error before the normal warning location then it should still fire 

```ruby
    @warn_outdated ||= begin
       @outdated_version_check.join

        warn_outdated_minor
       warn_outdated_eol
       true
     end
```

Reply:
👍